File: generate-icon.js
const fs = require('fs');
const { createCanvas } = require('canvas');

// Create 1024x1024 canvas
const canvas = createCanvas(1024, 1024);
const ctx = canvas.getContext('2d');

// Green background (#10b981 - emerald-500)
ctx.fillStyle = '#10b981';
ctx.fillRect(0, 0, 1024, 1024);

// Draw white mosque icon
ctx.fillStyle = '#ffffff';
ctx.strokeStyle = '#ffffff';
ctx.lineWidth = 8;

const centerX = 512;
const centerY = 512;
const scale = 1.5; // Make icon bigger and fitting

// Main dome (top center)
ctx.beginPath();
ctx.arc(centerX, centerY - 120 * scale, 80 * scale, Math.PI, 0, false);
ctx.fill();

// Crescent on top of dome
ctx.save();
ctx.translate(centerX, centerY - 200 * scale);
ctx.rotate(-Math.PI / 6);
ctx.beginPath();
ctx.arc(0, 0, 30 * scale, 0, Math.PI * 2);
ctx.fill();
ctx.beginPath();
ctx.fillStyle = '#10b981';
ctx.arc(15 * scale, 0, 30 * scale, 0, Math.PI * 2);
ctx.fill();
ctx.restore();
ctx.fillStyle = '#ffffff';

// Main building body
ctx.fillRect(centerX - 140 * scale, centerY - 40 * scale, 280 * scale, 180 * scale);

// Left minaret
ctx.fillRect(centerX - 200 * scale, centerY - 100 * scale, 40 * scale, 240 * scale);
// Left minaret dome
ctx.beginPath();
ctx.arc(centerX - 180 * scale, centerY - 100 * scale, 30 * scale, Math.PI, 0, false);
ctx.fill();
// Left minaret crescent
ctx.beginPath();
ctx.arc(centerX - 180 * scale, centerY - 140 * scale, 12 * scale, 0, Math.PI * 2);
ctx.fill();
ctx.beginPath();
ctx.fillStyle = '#10b981';
ctx.arc(centerX - 172 * scale, centerY - 140 * scale, 12 * scale, 0, Math.PI * 2);
ctx.fill();
ctx.fillStyle = '#ffffff';

// Right minaret
ctx.fillRect(centerX + 160 * scale, centerY - 100 * scale, 40 * scale, 240 * scale);
// Right minaret dome
ctx.beginPath();
ctx.arc(centerX + 180 * scale, centerY - 100 * scale, 30 * scale, Math.PI, 0, false);
ctx.fill();
// Right minaret crescent
ctx.beginPath();
ctx.arc(centerX + 180 * scale, centerY - 140 * scale, 12 * scale, 0, Math.PI * 2);
ctx.fill();
ctx.beginPath();
ctx.fillStyle = '#10b981';
ctx.arc(centerX + 188 * scale, centerY - 140 * scale, 12 * scale, 0, Math.PI * 2);
ctx.fill();
ctx.fillStyle = '#ffffff';

// Main entrance arch
ctx.beginPath();
ctx.arc(centerX, centerY + 60 * scale, 50 * scale, Math.PI, 0, false);
ctx.fillStyle = '#10b981';
ctx.fill();

// Door
ctx.fillStyle = '#10b981';
ctx.fillRect(centerX - 50 * scale, centerY + 60 * scale, 100 * scale, 80 * scale);

// Windows (decorative arches)
// Left window
ctx.fillStyle = '#10b981';
ctx.beginPath();
ctx.arc(centerX - 80 * scale, centerY, 25 * scale, Math.PI, 0, false);
ctx.fill();
ctx.fillRect(centerX - 105 * scale, centerY, 50 * scale, 40 * scale);

// Right window
ctx.beginPath();
ctx.arc(centerX + 80 * scale, centerY, 25 * scale, Math.PI, 0, false);
ctx.fill();
ctx.fillRect(centerX + 55 * scale, centerY, 50 * scale, 40 * scale);

// Save to resources/icon.png
const buffer = canvas.toBuffer('image/png');
fs.writeFileSync('resources/icon.png', buffer);
console.log('✅ Icon generated: resources/icon.png');
console.log('📱 Run: npx @capacitor/assets generate --android');

File: generate-icon.py
from PIL import Image, ImageDraw
import math

# Create 1024x1024 image with green background
size = 1024
img = Image.new('RGB', (size, size), color='#10b981')
draw = ImageDraw.Draw(img)

centerX = 512
centerY = 512
scale = 1.5

white = '#ffffff'
green = '#10b981'

# Helper function to draw filled arc (dome)
def draw_dome(draw, x, y, radius, color):
    # Draw semicircle as filled polygon
    points = []
    for angle in range(180, 361):
        rad = math.radians(angle)
        px = x + radius * math.cos(rad)
        py = y + radius * math.sin(rad)
        points.append((px, py))
    if points:
        draw.polygon(points, fill=color)

# Main dome
draw_dome(draw, centerX, centerY - 120 * scale, 80 * scale, white)

# Main building body
left = centerX - 140 * scale
top = centerY - 40 * scale
right = centerX + 140 * scale
bottom = centerY + 140 * scale
draw.rectangle([left, top, right, bottom], fill=white)

# Left minaret
left_min = centerX - 200 * scale
top_min = centerY - 100 * scale
right_min = centerX - 160 * scale
bottom_min = centerY + 140 * scale
draw.rectangle([left_min, top_min, right_min, bottom_min], fill=white)
# Left minaret dome
draw_dome(draw, centerX - 180 * scale, centerY - 100 * scale, 30 * scale, white)

# Right minaret
left_min_r = centerX + 160 * scale
right_min_r = centerX + 200 * scale
draw.rectangle([left_min_r, top_min, right_min_r, bottom_min], fill=white)
# Right minaret dome
draw_dome(draw, centerX + 180 * scale, centerY - 100 * scale, 30 * scale, white)

# Main entrance arch (green cutout)
draw_dome(draw, centerX, centerY + 60 * scale, 50 * scale, green)
# Door (green rectangle)
door_left = centerX - 50 * scale
door_top = centerY + 60 * scale
door_right = centerX + 50 * scale
door_bottom = centerY + 140 * scale
draw.rectangle([door_left, door_top, door_right, door_bottom], fill=green)

# Left window (green cutout)
draw_dome(draw, centerX - 80 * scale, centerY, 25 * scale, green)
win_left = centerX - 105 * scale
win_right = centerX - 55 * scale
win_top = centerY
win_bottom = centerY + 40 * scale
draw.rectangle([win_left, win_top, win_right, win_bottom], fill=green)

# Right window (green cutout)
draw_dome(draw, centerX + 80 * scale, centerY, 25 * scale, green)
win_left_r = centerX + 55 * scale
win_right_r = centerX + 105 * scale
draw.rectangle([win_left_r, win_top, win_right_r, win_bottom], fill=green)

# Crescents on domes (white circles with green overlay)
# Main dome crescent
crescent_main_x = centerX
crescent_main_y = centerY - 200 * scale
radius_crescent = 30 * scale
draw.ellipse([crescent_main_x - radius_crescent, crescent_main_y - radius_crescent,
              crescent_main_x + radius_crescent, crescent_main_y + radius_crescent], fill=white)
draw.ellipse([crescent_main_x - radius_crescent + 15 * scale, crescent_main_y - radius_crescent,
              crescent_main_x + radius_crescent + 15 * scale, crescent_main_y + radius_crescent], fill=green)

# Left minaret crescent
crescent_l_x = centerX - 180 * scale
crescent_l_y = centerY - 140 * scale
radius_small = 12 * scale
draw.ellipse([crescent_l_x - radius_small, crescent_l_y - radius_small,
              crescent_l_x + radius_small, crescent_l_y + radius_small], fill=white)
draw.ellipse([crescent_l_x - radius_small + 8 * scale, crescent_l_y - radius_small,
              crescent_l_x + radius_small + 8 * scale, crescent_l_y + radius_small], fill=green)

# Right minaret crescent
crescent_r_x = centerX + 180 * scale
draw.ellipse([crescent_r_x - radius_small, crescent_l_y - radius_small,
              crescent_r_x + radius_small, crescent_l_y + radius_small], fill=white)
draw.ellipse([crescent_r_x - radius_small + 8 * scale, crescent_l_y - radius_small,
              crescent_r_x + radius_small + 8 * scale, crescent_l_y + radius_small], fill=green)

# Save image
img.save('resources/icon.png')
print('✅ Icon generated: resources/icon.png')
print('📱 Run: npx @capacitor/assets generate --android')
